sdk/opendp/whitenoise/data/dpsu.py

```diff
@@ -0,0 +1,117 @@
+import pandas as pd
+import math
+import random
+import operator
+from collections import defaultdict
+
+from opendp.whitenoise.metadata import CollectionMetadata
+from opendp.whitenoise.sql import PandasReader
+from opendp.whitenoise.sql.parse import QueryParser
+from opendp.whitenoise.mechanisms.rand import laplace
+
+def preprocess_df_from_query(meta_path, csv_path, query, table_name):
```

Comment:
inputs should be schema, pandas dataframe

It is the same but a bit more flexible since it won't require users to write their in memory objects to disk. It is also a one liner(as shown here) to read it from disk so it isn't too cumbersome